tests/storage/reservation.go

```diff
@@ -0,0 +1,295 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/rand"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/exec"
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/libstorage"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+// The SCSI persistent reservation tests require to run serially because of the
+// feature gate PersistentReservation. The enablement/disablement of this
+// feature gate redeploys virt-handler pod, and this might interfer with other
+// tests.
+var _ = SIGDescribe("[Serial]SCSI persistent reservation", Serial, func() {
+	const randLen = 8
+	var (
+		naa          string
+		backendDisk  string
+		disk         string
+		targetCliPod string
+		virtClient   kubecli.KubevirtClient
+		node         string
+		device       string
+		fgDisabled   bool
+		pv           *corev1.PersistentVolume
+		pvc          *corev1.PersistentVolumeClaim
+	)
+
+	// NAA is the Network Address Authority and it is an identifier represented
+	// in ASCII-encoded hexadecimal digits
+	// More details at:
+	//  https://datatracker.ietf.org/doc/html/rfc3980#ref-FC-FS
+	generateNaa := func() string {
+		const letterBytes = "0123456789abcdef"
+		b := make([]byte, 14)
+		for i := range b {
+			b[i] = letterBytes[rand.Intn(len(letterBytes))]
+		}
+		// Keep the first 2 digits constants as not all combinations are valid naa
+		return "52" + string(b)
+	}
+
+	// executeTargetCli executes command targetcli
+	executeTargetCli := func(podName string, args []string) {
+		cmd := append([]string{"/usr/bin/targetcli"}, args...)
+		pod, err := virtClient.CoreV1().Pods(testsuite.NamespacePrivileged).Get(context.Background(), podName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		stdout, stderr, err := exec.ExecuteCommandOnPodWithResults(virtClient, pod, "targetcli", cmd)
+		By(fmt.Sprintf("targetcli: stdout: %v stderr: %v", stdout, stderr))
```

Comment:
This is miss use of `By`. You probably want `Expect(err).ToNot(HaveOccurred(), stdout, stderr)`